File: src/main/java/common/item/ItemIngot.java
package common.item;

import java.awt.*;

public class ItemIngot extends ItemColored {
    private int color;

    public ItemIngot(String name, int color) {
        super(name);
        this.color = color;
    }

    public ItemIngot(String name) {
        super(name);
        this.color = 0xFFF;
    }

    @Override
    public void onItemUse() { }

    public Color getColor() {
        return  new Color(color);
    }
}

File: src/main/kotlin/common/item/ItemIngot.kt
package common.item

import java.awt.*

class ItemIngot : ItemColored {
    private var color: Int = 0

    constructor(name: String, color: Int) : super(name) {
        this.color = color
    }

    constructor(name: String) : super(name) {
        this.color = 0xFFF
    }

    override fun onItemUse() {}

    fun getColor(): Color {
        return Color(color)
    }
}
